src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs

```diff
@@ -1141,35 +1141,15 @@ public static CultureInfo GetCultureInfo(string name, bool predefinedOnly)
             return GetCultureInfo(name);
         }
 
-        private static Dictionary<string, CultureInfo> CachedCulturesByName
-        {
-            get
-            {
-                Dictionary<string, CultureInfo>? cache = s_cachedCulturesByName;
-                if (cache is null)
-                {
-                    cache = new Dictionary<string, CultureInfo>();
-                    cache = Interlocked.CompareExchange(ref s_cachedCulturesByName, cache, null) ?? cache;
-                }
-
-                return cache;
-            }
-        }
+        private static Dictionary<string, CultureInfo> CachedCulturesByName =>
+            s_cachedCulturesByName ??
```

Comment:
Does this introduce a race condition where the property could return `null` unexpectedly?

https://github.com/dotnet/runtime/blob/776ab3cd89e1622fe767710f021ab1e86cbdeba3/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs#L782

Reply:
Yes, if it can be nulled out concurrently, after this compare exchange sets it, it's possible that could cause this to return null.

Reply:
The existing code does not have this race condition. It sounds like this refactoring should be reverted.

Reply:
Yes, I plan to fix it, I'm just not near my computer that.

Reply:
Fixed